types/ecore/index.d.ts

```diff
@@ -0,0 +1,151 @@
+        array: () => EObject[];
+        first: () => EObject;
+        find: (iterator: (value: any, key: any, list: EList) => boolean, context?: any) => EObject[];
+        last: () => EObject;
+        rest: (position: number) => EObject[];
+        each: (iterator: (value: any, key: any, list: EList) => void, context?: any) => void;
+        filter: (iterator: (value: any, key: any, list: EList) => boolean, context?: any) => EObject[];
+        map: (iterator: (value: any, key: any, list: EList) => any, context?: any) => any[];
+        reject: (iterator: (value: any, key: any, list: EList) => boolean, context?: any) => EObject[];
+        contains: (eObject: EObject) => boolean;
+        indexOf: (eObject: EObject) => number;
+    }
+    type EString = EObject;
+    type EInt = EObject;
+    type EBoolean = EObject;
+    type EDouble = EObject;
+    type EDate = EObject;
+    type EIntegerObject = EObject;
+    type EFloatObject = EObject;
+    type ELongObject = EObject;
+    type EMap = EObject;
+    type EDiagnosticChain = EObject;
+    type JSObject = EObject;
+    type EModelElement = EObject;
+    type EAnnotation = EModelElement;
+    type ENamedElement = EModelElement;
+    type EClassifier = ENamedElement;
+    type EClass = EClassifier;
+    type EDataType = EClassifier;
+    type ETypedElement = ENamedElement;
+    type EStructuralFeature = ETypedElement;
+    type EAttribute = EStructuralFeature;
+    type EReference = EStructuralFeature;
+    type EOperation = ETypedElement;
+    type EParameter = ETypedElement;
+    type EEnum = EDataType;
+    type EEnumLiteral = ENamedElement;
+    type EGenericType = EObject;
+    type ETypeParameter = ENamedElement;
+    interface Resource extends EObject {
+        add: (value: EObject) => void;
+        addAll: (values: EObject[]) => EObject;
+        clear: () => EList;
+        each: (iterator: (value: any, key: any, list: EList) => void, context?: any) => void;
+        save: (callback: () => void, options: any) => void;
+        parse: (data: EObject, loader: () => void) => any;
+        remove: () => void;
+        rev: string;
+        load: (res: any) => void;
+        to: () => any;
+    }
+    interface EPackage extends EObject {
+        getEPackage: (nsURI: string) => EPackage;
+        register: (ePackage: EPackage) => void;
+        ePackages: () => EPackage[];
+        elements: (type: EObject) => EObject[];
+        Registry: EPackageRegistry;
+    }
+    interface EPackageRegistry extends EObject {
+        register: (ePackage: EPackage) => void;
+        ePackages: () => EPackage[];
+    }
+    interface ResourceSet extends EObject {
+        create: (uri: any) => Resource;
+        getEObject: (uri: string) => EObject;
+        parse: (data: EObject) => void;
+        elements: (type?: string | EClass) => EObject[];
+        toJSON: () => any;
+    }
+    namespace ResourceSet {
+        function create(): ResourceSet;
+    }
+}
+
+export default Ecore;
```

Comment:
ecore's source package doesn't have a default export as far as I can tell. I think the correct syntax is the commonjs export: `export = Ecore;`

Reply:
When I do it like that, I get errors during tests.

Error: F:/git/types/DefinitelyTyped/types/ecore/ecore-tests.ts:1:10
ERROR: 1:10   expect                 TypeScript@next compile error: 
Module '"../../../DefinitelyTyped/types/ecore"' has no exported member 'Ecore'.

F:/git/types/DefinitelyTyped/types/ecore/index.d.ts:151:1
ERROR: 151:1  export-just-namespace  Instead of `export =`-ing a namespace, use the body of the namespace as the module body. See: https://github.com/Microsoft/dtslint/blob/master/docs/export-just-namespace.md

Reply:
I'm sorry, I forgot to mention that the import also needs to match the export: `import Ecore = require('ecore')`.

`import { Ecore } from 'ecore'` would technically work but only because the package source does `Ecore.Ecore = Ecore` (or its equivalent).

Reply:
Thank you! I've fixed the first error, but still, the second error is not gone. What can I do with that?

```
Error: F:/git/types/DefinitelyTyped/types/ecore/index.d.ts:151:1
ERROR: 151:1  export-just-namespace  Instead of `export =`-ing a namespace, use the body of the namespace as the module body. See: https://github.com/Microsoft/dtslint/blob/master/docs/export-just-namespace.md
```

Reply:
Oops. I didn't read carefully enough to see that error. That lint rule is asking you to delete `declare namespace Ecore` and `export = Ecore`, dedent the body of the namespace, and put the `export` keyword in front of anything that should be exported (which is probably everything). They are equivalent, plus it allows people to directly import a few symbols from the module if they want.

Reply:
Well, I did some research. As far as I can understand I had to get rid of a namespace declaration, cause in case of 'require' it would look strange like Ecore.Ecore as you said. I don't know whether it an appropriate way or not.

Reply:
Looks good to me now.

Reply:
Thank you for your help and kindness!